Update workspace creation documentation (#17)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/creating.mdx

1. Click **Workspaces** in the top navigation bar to view a list of the workspaces within your organization.
1. Click **+ New Workspace**. The **Create a new Workspace** page appears.
1. Choose a workflow type: **Version control**, **CLI-driven**, or **API-driven**.
1. (Version control only) Choose an existing version control provider or configure a new one. Refer to [Connecting VCS Providers](/cloud-docs/vcs) for more details.
1. (Version control only) Choose a repository from the filterable list. The **Configure settings** page appears.

-> **Note:** Depending on your provider, you may need to change which account's repositories are shown. You can also scroll to the bottom of the list and enter the repository ID in the text field if it is not immediately visible.

1. Enter a **Workspace Name**. This defaults to the repository name, if applicable. The name must be unique within the organization and can include letters, numbers, dashes (`-`), and underscores (`_`). Refer to our [workspace naming recommendations](/cloud-docs/workspaces/naming).
1. Add an optional **Description** that will appear at the top of the workspace in the Terraform Cloud UI.

> 1. (Version control only) Open **Advanced options** to optionally configure the following additional settings:

Reviewer: ```suggestion
1. (Version control only) Open **Advanced options** to optionally configure the following settings:
```

   - [Terraform Working Directory][workdir]
   - [Automatic Run Triggering][trigger]
   - [VCS branch][branch]
   - [Include submodules on clone][submodules]

1. Click **Create workspace**.

For CLI and API-driven workspaces, the workspace overview page appears. For version control workspaces, the **Configure Terraform variables** page appears.

### Configure Terraform Variables (Version Control Only)

> After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) and displays them on the page. Terraform variables without default values or variables not defined by a [Variable Set](/cloud-docs/api-docs/variable-sets) will be displayed. Terraform cannot perform successful runs until you set values for these variables.

Reviewer: ```suggestion
After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) and displays any that do not have a default value and are not defined in an existing global [variable set](/cloud-docs/api-docs/variable-sets). Terraform cannot perform successful runs in the workspace until you set values for these variables. 
```

Choose one of the following actions:

> - To skip this step, click **Go to workspace overview**. You can create these variables later from within the workspace.

Reviewer: ```suggestion
- To skip this step, click **Go to workspace overview**. You can [load these variables from files](/cloud-docs/workspaces/variables/managing-variables#loading-variables-from-files) or create and set values for them later from within the workspace. Terraform Cloud will not automatically scan your configuration again; you can only add variables from within the workspace individually. 
```

> - To create variables, enter a value for each variable. If a value is provided by another source (e.g., `auto.tfvars` file), click **Ignore variable**. Click **Save variables**.

Reviewer: ```suggestion
- To configure variables, enter a value or click **Ignore variable** for each variable on the page. You may want to ignore a variable if you know you will provide it through another source, like an `auto.tfvars` file. Click **Save variables** to add these variables to the workspace.
```

Author: Yeeeeees, I love this.

- [Upload configuration versions](/cloud-docs/workspaces/configurations#providing-configuration-versions): If you chose the API or CLI-Driven workflow, you must upload configuration versions for the workspace.
- [Edit environment variables](/cloud-docs/workspaces/variables): Shell environment variables store credentials and customize Terraform's behavior.
- [Edit additional workspace settings](/cloud-docs/workspaces/settings): This includes notifications, permissions, and run triggers to start runs automatically.
- [Learn more about running Terraform in your workspace](/cloud-docs/run): This includes how Terraform processes runs within the workspace, run modes, run states, etc.

### VCS Connection

If you connected a VCS repository to the workspace, Terraform Cloud automatically registers a webhook with your VCS provider. A workspace with no runs will not accept new runs from a VCS webhook, so you must [manually start at least one run](/cloud-docs/run/ui#manually-starting-runs).

> After you have manually started a run, Terraform Cloud will automatically queue a plan for the workspace when new commits appear in the selected branch of the linked repository or someone opens a pull request on that branch. Refer to [Webhooks](/cloud-docs/vcs/#webhooks) for more details.

Reviewer: ```suggestion
After you manually start a run, Terraform Cloud automatically queues a plan when new commits appear in the selected branch of the linked repository or someone opens a pull request on that branch. Refer to [Webhooks](/cloud-docs/vcs/#webhooks) for more details.
```

-> **Note:** Depending on your provider, you may need to change which account's repositories are shown. If the repository is not listed, scroll to the bottom of the list and enter its ID in the text field.

1. Enter a **Workspace Name**. This defaults to the repository name, if applicable. The name must be unique within the organization and can include letters, numbers, dashes (`-`), and underscores (`_`). Refer to our [workspace naming recommendations](/cloud-docs/workspaces/naming).
1. Add an optional **Description** that will appear at the top of the workspace in the Terraform Cloud UI.

1. (Version control only) Open **Advanced options** to optionally configure the following additional settings:
   - [Terraform Working Directory][workdir]
   - [Automatic Run Triggering][trigger]
   - [VCS branch][branch]
   - [Include submodules on clone][submodules]

1. Click **Create workspace**.

For CLI and API-driven workspaces, the workspace overview page appears. For version control workspaces, the **Configure Terraform variables** page appears.

### Configure Terraform Variables (Version Control Only)

> After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) without default values and displays them on the page. Terraform cannot perform successful runs until you set values for these variables.

Reviewer 2: Might be worth pointing out that variables that are in varsets applied to the workspace will also not show up in the form.

```suggestion
After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) that are not present in applicable varsets or do not have default values and displays them on the page. Terraform cannot perform successful runs until you set values for these variables.
```

Reviewer 2: Might also be worth mentioning that this does not include environment variables either?

Author: I thought it would be unnecessary because Terraform variables and environment variables are different types of variables (it's clarified in the Terraform variable link). If we say Terraform variable, will it be clear that we mean Terraform variables, not any type of variable? I thought it would be but totally open to changing.

Author: Oooo I didn't think of varsets - can those be set in a configuration file and we just ignore it during ingress?

> -> **Note:** Your Terraform configuration needs to be in the root (default) working directory for this to work. If you're using a different working directory, configure variables from within the workspace.

Reviewer 2: this is no longer true.
The variable detection is possible in given (non-default) working directories.

Author: Awesome, I will remove!
